Adds events-at-cu.py, which pulls events from events.columbia.edu and adds them to the database.

tarapichumani opened the pull request.

tarapichumani committed:
Adds events-at-cu.py.

alanhdu commented on cal/events-at-cu.py:
```diff
@@ -0,0 +1,38 @@
+def update_from_eventsatcu():
+
+
+    url = "https://events.columbia.edu/feeder/main/eventsFeed.gdo?f=y&sort=dtstart.utc:asc&fexpr=(categories.href!=%22/public/.bedework/categories/sys/Ongoing%22)%20and%20(categories.href=%22/public/.bedework/categories/org/UniversityEvents%22)%20and%20(entity_type=%22event%22%7Centity_type=%22todo%22)&format=text/calendar&count=200"
```
This is a really long URL. Can we shorten it?

tarapichumani committed:
style issues (shortening url, whitespace)

alanhdu commented on cal/events-at-cu.py:
```diff
@@ -0,0 +1,31 @@
+
+def update_from_eventsatcu():
+
+    url = "https://events.columbia.edu/feeder/main/eventsFeed.gdo?f=y&sort=dtstart.utc:asc&format\=text/calendar&count=200"
```
I think there's an extra `\` that's not needed: 

```
url = "https://events.columbia.edu/feeder/main/eventsFeed.gdo?f=y&sort=dtstart.utc:asc&format\=text/calendar&count=200"
                                                                                             ^
```

This is also still a little long. Maybe we should do something like:

``` Python
url = "https://evens.columbia.edu/feeder/main/eventsFeed.gdo" + \
       "?f=y&sort=dtstart.utc:asc&format=text/calendar&count=200"
```

Could you also document why we need each part of the query string does? (For example, why do we need to specify `&count=200`? What's going on with the `f=y`?).

alanhdu commented:
Flake8 output:

```
./cal/events-at-cu.py:7:80: E501 line too long (123 > 79 characters)
./cal/events-at-cu.py:10:73: W291 trailing whitespace
./cal/events-at-cu.py:15:9: F841 local variable 'event_url' is assigned to but never used
./cal/events-at-cu.py:19:32: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:19:34: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:19:50: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:19:52: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:19:68: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:19:70: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:19:80: E501 line too long (98 > 79 characters)
./cal/events-at-cu.py:19:88: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:19:90: E251 unexpected spaces around keyword / parameter equals
./cal/events-at-cu.py:31:1: W391 blank line at end of file
```

alanhdu commented on cal/events-at-cu.py:
```diff
@@ -0,0 +1,31 @@
+
+        cevent = Event.query.filter(Event.sundial_id == event_id).first()
+        if cevent is None:
+            cevent = Event(name = event_name, url = url, sundial_id = event_id, user_id = user.id)
```
If this line is too long, we should break it up like:

``` Python
cevent = Event(name=event_name, url=url,
               sundial_id=event_id, user_id=user.id)
```

tarapichumani committed:
Update url

- remove extraneous 'f=y'
- count=200 necessary to return max possible events
- fixed typo

tarapichumani committed:
Document query string

tarapichumani committed:
fixed flake errors, other formatting issues.
